src/debug/telemetry/telemetry2_thread_info.c

```diff
@@ -0,0 +1,327 @@
+/* SPDX-License-Identifier: Apache-2.0
+ * Copyright (c) 2024 Intel Corporation
+ */
+
+#include <zephyr/logging/log.h>
+#include <zephyr/kernel.h>
+#include <stdio.h>
+#include <soc.h>
+#include <adsp_debug_window.h>
+
+#include "telemetry2_slot.h"
+
+LOG_MODULE_REGISTER(thread_info);
+
+/* to adsp backend header, data to be shared with user space client*/
+#define THREAD_INFO_MAX_THREADS 16
+
+struct thread_info {
+	char name[14];
+	uint8_t stack_usage;	/* precentage */
+	uint8_t cpu_usage;	/* precentage */
+} __packed;
+
+#define THREAD_INFO_VERSION_0_0		0
+
+#define THREAD_INFO_STATE_UNINITIALIZED	0
+#define THREAD_INFO_STATE_BEING_UPDATED	1
+#define THREAD_INFO_STATE_UPTODATE	2
+
+struct thread_info_core {
+	uint8_t state;
+	uint8_t counter;		/* incremented every round */
+	uint8_t load;			/* precentage */
+	uint8_t thread_count;
+	struct thread_info thread[THREAD_INFO_MAX_THREADS];
+} __packed __aligned(64);
+
+struct thread_info_chunk {
+	struct telemetry2_chunk_hdr hdr;
+	uint8_t core_count;
```

Comment:
uint32_t as the core_offset array will be packed and aligned on a byte

Reply:
Oh yes. I used to have more fields here, but dropped them. I'll make core_count uint16_t too. Not that it makes much difference as the core_offsets are written DSP side only once.